sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -64,18 +63,28 @@ import org.apache.spark.util.Utils
  *
  * @since 1.0.0
  */
-class SQLContext(@transient val sparkContext: SparkContext)
-  extends org.apache.spark.Logging
-  with Serializable {
+class SQLContext private[sql](
+    @transient val sparkContext: SparkContext,
+    @transient protected[sql] val cacheManager: CacheManager)
+  extends org.apache.spark.Logging with Serializable {
 
   self =>
 
+  def this(sparkContext: SparkContext) = this(sparkContext, new CacheManager)
```

Comment:
Let's add scaladoc to explain when it is recommended to use this constructor and when it recommended to use `newSession`.